cse/src/main/java/com/adyen/checkout/cse/GenericEncrypter.kt

```diff
@@ -0,0 +1,44 @@
+package com.adyen.checkout.cse
+
+import com.adyen.checkout.cse.exception.EncryptionException
+import org.json.JSONException
+import org.json.JSONObject
+import java.text.SimpleDateFormat
+import java.util.*
+
+object GenericEncrypter {
+
+    const val ENCRYPTION_FAILED_MESSAGE = "Encryption failed."
+    private val GENERATION_DATE_FORMAT = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.US).apply {
+        timeZone = TimeZone.getTimeZone("UTC")
+    }
+
+    const val KCP_PASSWORD_KEY = "password"
+
+    @JvmStatic
+    fun encryptField(
+        encryptionKey: String,
+        fieldToEncrypt: Any,
+        publicKey: String,
+        generationTime: String = makeGenerationTime()
```

Comment:
I think `generationTime` was not a parameter in then CardEncrypter on purpose so that the encryption method will always take the current system time to make a bit harder to "fake".